cmake/compiler/gcc/compiler_flags.cmake

```diff
@@ -200,3 +200,7 @@ set_compiler_property(PROPERTY no_position_independent
                       -fno-pic
                       -fno-pie
 )
+
+# Compiler flag to instruct the assembler to translate direct calls to
+# indirect calls unless it can determine that the target of a direct call.
+set_compiler_property(PROPERTY long_calls "-mlongcalls")
```

Comment:
this appears to be an Xtensa specific option.

Coupled with other comment where old code would go through a chain of commands that would include a `zephyr_check_compiler_flag()` I believe this probably should be adjusted to a `check_set_compiler_property()` call.